app/apphandlers/setup.go

```diff
@@ -222,3 +225,33 @@ func monitorDestRouters(ctx context.Context, routerFactory *router.Factory, batc
 	}
 	wg.Wait()
 }
+
+// NewRsourcesService produces a rsources.JobService through environment configuration (env variables & config file)
+func NewRsourcesService(deploymentType deployment.Type) (rsources.JobService, error) {
+	var rsourcesConfig rsources.JobServiceConfig
+	rsourcesConfig.MaxPoolSize = config.GetInt("Rsources.PoolSize", 5)
+	rsourcesConfig.LocalConn = jobsdb.GetConnectionString()
+	rsourcesConfig.LocalHostname = config.GetEnv("JOBS_DB_HOST", "localhost")
+
+	sharedConn := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable application_name=%s",
+		config.GetEnv("SHARED_DB_HOST", "undefined"),
+		config.GetEnv("SHARED_DB_PORT", "undefined"),
+		config.GetEnv("SHARED_DB_USER", "undefined"),
+		config.GetEnv("SHARED_DB_PASSWORD", "undefined"),
+		config.GetEnv("SHARED_DB_DB_NAME", "undefined"),
+		misc.DefaultString("rudder-server").OnError(os.Hostname()))
```

Comment:
I know that we are going to break consistency, but I would advocate using `SHARED_DB_DSN` - a single env variable to define the SQL connection